ext/erb/erb.c

```diff
@@ -74,7 +76,9 @@ optimized_escape_html(VALUE str)
 static VALUE
 erb_escape_html(VALUE self, VALUE str)
 {
-    str = rb_convert_type(str, T_STRING, "String", "to_s");
+    if (!RB_TYPE_P(str, T_STRING)) {
+        str = rb_convert_type(str, T_STRING, "String", "to_s");
```

Comment:
Why not `rb_check_string_type`?

Reply:
It calls `to_str` instead of `to_s`, right? Almost every template engine calls `to_s` instead of `to_str` before passing an object to an HTML escape method, so it needs to be `to_s`. Note that the previous Ruby implementation of `ERB::Util.html_escape` called `to_s` before passing an argument to `CGI.escapeHTML` too.